util/GameRuleRanks.h

```diff
@@ -0,0 +1,54 @@
+#ifndef _GameRuleRanks_h_
+#define _GameRuleRanks_h_
+
+/** This file is to keep internal engine game rules' ranks in one place for easier re-ordering,
+    though still needs to be manually synchronized with game_rules.focs.py
+    */
+
+namespace GameRuleRanks {
+	enum GameRuleRanks : uint32_t {
+
+		// "" aka "GENERAL" category
+		RULE_RESEED_PRNG_SERVER_RANK = 1,
+		RULE_EXTRASOLAR_SHIP_DETECTION_RANK = 2,
+		RULE_BASIC_VIS_SYSTEM_INFO_SHOWN_RANK = 10,
+		RULE_NUM_COMBAT_ROUNDS_RANK = 30,
+		RULE_AGGRESSIVE_SHIPS_COMBAT_VISIBLE_RANK = 35,
+		RULE_STOCKPILE_IMPORT_LIMITED_RANK = 50,
+		RULE_PRODUCTION_QUEUE_FRONTLOAD_FACTOR_RANK = 51,
+		RULE_PRODUCTION_QUEUE_TOPPING_UP_FACTOR_RANK = 52,
+
+		// "BALANCE" category
+		RULE_SHIP_WEAPON_DAMAGE_FACTOR_RANK = 331,
+		RULE_FIGHTER_DAMAGE_FACTOR_RANK = 332,
+		RULE_SHIP_STRUCTURE_FACTOR_RANK = 333,
+		RULE_SHIP_SPEED_FACTOR_RANK = 334,
+
+		// "TEST" category
+		RULE_CHEAP_AND_FAST_BUILDING_PRODUCTION_RANK = 1010,
+		RULE_CHEAP_AND_FAST_SHIP_PRODUCTION_RANK = 1011,
+		RULE_CHEAP_AND_FAST_TECH_RESEARCH_RANK = 1012,
+		RULE_CHEAP_POLICIES_RANK = 1013,
+		RULE_ALL_OBJECTS_VISIBLE_RANK = 1030,
+		RULE_ALL_SYSTEMS_VISIBLE_RANK = 1031,
+		RULE_UNSEEN_STEALTHY_PLANETS_INVISIBLE_RANK = 1032,
+		RULE_STARLANES_EVERYWHERE_RANK = 1035,
+
+		// "BALANCE_STABILITY" category (should there be a separate opinion and annexation group?)
+		RULE_ANNEX_COST_OPINION_EXP_BASE_RANK = 5500,
+		RULE_ANNEX_COST_STABILITY_EXP_BASE_RANK = 5510,
```

Comment:
replace all the tabs with 4 or 8 spaces
```suggestion
                RULE_ANNEX_COST_STABILITY_EXP_BASE_RANK = 5510,
```